### ISSUE TITLE
Author's Username is missing from Author dropdown.

### ISSUE THREAD

**Issue by TwisterMc** [WordPress/gutenberg#17364]
**Describe the bug**
In the classic editor, the author drop down included their name followed by their username in parentheses. This is needed when there are two or more people with the same name, or one user that has different kinds of access for testing things. Gutenberg however only shows the author's name.

**To reproduce**
1. Create a new post using the Classic editor. 
2. Click on the author drop down and you'll see users in this format: Name (username)
3. Create a new post using the Gutenberg editor. 
4. Click on the author drop down and you'll see users in this format: Name

**Expected behavior**
 I'd see users in this format: Name (username)

**Screenshots**
![Gutenberg](https://user-images.githubusercontent.com/46997/64461068-bf676280-d0c1-11e9-9214-844bd4c307d7.png)
![Classic Editor](https://user-images.githubusercontent.com/46997/64461081-ca21f780-d0c1-11e9-9e02-803c4fe97514.png)
![Quick Edit Post](https://user-images.githubusercontent.com/46997/64461089-d1e19c00-d0c1-11e9-902e-aecf28b85242.png)

**Desktop (please complete the following information):**
 - OS: MacOS 10.14.6 (18G95)
 - Browser Safari / Firefox / Chrome
 - Basically everything as it's a WordPress concern.

**Additional context**
I marked this as a bug because the author drop down is consistent across the backend, except for in Gutenberg. If you were to quick edit a post or page, the format would match what I'm expecting. Gutenberg seems to be breaking the consistency.

Additionally my client has nearly 150 users so what we have today in Gutenberg doesn't scale well if we just assume there will never be two users with the same name.

I've also dug for hooks or filters to try and customize what appears in the author drop down but have had no luck.

**Comment by draganescu** [comment]
The REST API response to `/wp/v2/users/&who=authors&per_page=100&_locale=user` does not return `username` so that should be augmented 1st.

cc @spacedmonkey do you have any ideas here other than just adding a new field in the response with the `username`? I ask because `slug` already "seems" to be the username.

**Comment by noisysocks** [comment]
There's a new design by @javierarce for the _Author_ dropdown in https://github.com/WordPress/gutenberg/pull/41430#issuecomment-1157690629. Perhaps we could include username in the line underneath the display name alongside the user's role?

It's worth noting is that users are able to select how they want their name to appear in profile settings. This should be respected.

<img width="485" alt="Screen Shot 2022-07-15 at 13 14 23" src="https://user-images.githubusercontent.com/612155/179142790-91de65a3-c64e-4c61-b8f2-d25305c0951a.png">

**Comment by yanksyoon** [comment]
Hello! I am having the same issue with truncated response from the API - would there be a way to search by username only rather than including user's email in the search as well?
Thank you!

**Comment by draganescu** [comment]
My [previous comment](https://github.com/WordPress/gutenberg/issues/17364#issuecomment-618513761) is actually wrong. It appears that the endpoint is ready to return `username`, if it's present in the `_fields` param and the `context` is `edit`.

Currently the post authhor component in editor:

- does not request username 
- sets the context to view

The context on request has to be view so that the list of authors can be retrieved even if the logged in user does not have edit context caps.

@swissspidy @TimothyBJacobs could you shed some light why on the user schema `username` is restricted to the `edit` context? It seems from the OP that the data _is visible_ in author lists in other WP editors. Would you advise this control callse with an `edit` context just to get the username or change the schema to allow username (which is a user detail) be available in the `view` context.

I remember seeing people wanting to hide usernames for security reasons so I will assume the solution here is to:

- add the username to the fields requested by the control
- change the context to be `edit`, not sure what the harm here would be

**Comment by TimothyBJacobs** [comment]
Yeah, the `username` is not intended to be a public field so it's only available in `edit`. There is no cap like `view_usernames`, so we have to be more blunt. Fundamentally, we can't be as loosey goosey as a PHP interface (so to speak), because we aren't building bespoke endpoints for each interface.

So I'm not sure what the right solve here is, or if there even is a solve.

We could have a larger conversation around introducing a context like `block-editor`. That would be available to any user who can use a block editor. And then potentially expose the `username` field in the `block-editor` context. But even that isn't so obvious, for instance what happens with the "Block Editor Everywhere" project. It'd probably be unexpected that all form users on w.org could see usernames.

**Comment by draganescu** [comment]
Maybe then only users who can request the `edit` context will see the username like the OP exemplifies. Like in the editor if the logged in user is admin we request with `edit` and if not we request with `view` and no username appears in the control 🤷🏻

**Comment by TimothyBJacobs** [comment]
Yeah I think that would be where I lean to for a practical solve. The client can know whether the user has edit permissions, and can alter it's request appropriately.

**Comment by abhi3315** [comment]
I was thinking about whether there should be an autocomplete feature to select authors. It could be very useful, especially if there are a lot of authors on a site. Although there are many plugins available to handle this, such as [Co-Author Plus](https://wordpress.org/plugins/co-authors-plus/), having an autocomplete feature to list authors would make sense.

**Comment by Mamaduka** [comment]
The user autocompleter adds user slug as a suffix, and data is available in `view` context for logged-in users. Do you think we should use the same data here as well?

## Screenshot

![Image](https://github.com/user-attachments/assets/1ca7bc54-6f7c-463e-90a8-25e4016b7abe)

**Comment by coderGtm** [comment]
Hi everyone...I'm new here and was reading this thread.

The solution proposed by @Mamaduka seems like a good approach to me. Since the slug is public and also unique, plus it's available in the view context also, I think it's a practical solution.

I'd like to contribute if there's consensus.

Thanks...

**Comment by Mamaduka** [comment]
> So absent an easy way to surface the username slug, what we could do is add help-text to the author flyout. Something like:

Coming back from PR (#82019), I'm not sure whether "help-text" is actually useful. A user can only change their own display name settings, which is less helpful in lists.

**Comment by jasmussen** [comment]
Should we then close this issue, or is there still some value in #82019?

**Comment by Mamaduka** [comment]
It may be useful to display `user.slug` as a suffix in the dropdown, similar to the user autocompleter.

I plan to refactor the component as part of https://github.com/WordPress/gutenberg/issues/76986#issuecomment-5357920017, and we could try it there.

I think we can close https://github.com/WordPress/gutenberg/pull/82019.
